modules/openapi-generator/src/main/resources/dart2/api.mustache

```diff
@@ -165,7 +165,12 @@ class {{{classname}}} {
     if (response.statusCode >= 400) {
       throw ApiException(response.statusCode, _decodeBodyBytes(response));
     }
-    if (response.body != null) {
+
+    // When a remote server returns no body with a status of 204, we shall not decode it.
+    // At the time of writing this, `dart:convert` will throw an "Unexpected end of input"
+    // FormatException when trying to decode an empty string.
+    final responseBody = response.body;
+    if (responseBody != null && (responseBody.isNotEmpty || 204 != response.statusCode)) {
```

Comment:
lol you like yoga syntax :laughing: 

can we put the 204 in a const variable with proper name instead creating a magic number?

Reply:
or even better, an interface/class with such magic numbers we can refer to in the future

class HttpCodes {
int NO_BODY = 204;
}

That's probably more effort than you were aiming here, but I feel it would be an investment into code quality for future.

Reply:
Oops, yes yes I do :) I'll do that.

Reply:
There already exists such a class in http/html package: `HttpStatus`. Refactoring...